PURL-SPECIFICATION.rst

```diff
@@ -206,44 +206,63 @@ The rules for each component are:
     in the canonical form
   - Each ``subpath`` segment MUST be a percent-encoded string
   - When percent-decoded, a segment:
+
     - MUST NOT contain a '/'
     - MUST NOT be any of '..' or '.'
     - MUST NOT be empty
+
   - The ``subpath`` MUST be interpreted as relative to the root of the package
 
 
 Character encoding
 ~~~~~~~~~~~~~~~~~~
 
-For clarity and simplicity a ``purl`` is always an ASCII string. To ensure that
+For clarity and simplicity, a ``purl`` is always an ASCII string. To ensure that
```

Comment:
I would drop "For clarity and simplicity" which is a color commentary that I should not have included in the first place.

Then what about this:
```
A canonical ``purl`` is always an ASCII string composed only of these characters:
``A to Z``, ``a to z``, ``0 to 9`` and these punctuation ``:/@?#%.-_~``
```

... and then we can go on to specify the arcane and bizarre rules of percent encoding for purls for other characters, and that each component defines when and how to use the encoding defined here.

(And we likely should add a reference section further down with something like in RFC3986
   [ASCII]    American National Standards Institute, "Coded Character
              Set -- 7-bit American Standard Code for Information
              Interchange", ANSI X3.4, 1986.

)

Reply:
@pombredanne I've made these suggested updates as well, except for the detailed work covered by your phrase `and then we can go on to specify the arcane and bizarre rules of percent encoding for purls for other characters`.  We need (and don't yet have) agreement on these details before I can reduce them to text.

I see that you made no changes to the PR's current default of `MUST be percent-encoded unless . . ..`  I'm leaving that as is in the next iteration (which I'll push soon), but I have a suggested change to that default language that I'll add in a new comment once I've pushed that update.